Jenkinsfile

```diff
@@ -0,0 +1,63 @@
+#!/bin/env groovy
+
+node('us-east-1 && ubuntu && docker && !gpu') {
+
+    stage('Checkout') {
+        checkout scm
+    }
+    sh "`aws ecr get-login --region=us-east-1 --no-include-email`"
+    def dockerfileChecksum = sh(returnStdout: true, script: 'md5sum Dockerfile | cut -d" " -f1').trim()
+    def dockerTag = "${dockerfileChecksum}"
+    def baseImageName = "browser-f/android:${dockerTag}"
+    stage('Build docker image') {
+        docker.withRegistry('https://141047255820.dkr.ecr.us-east-1.amazonaws.com'){
+                try {
+                    def image = docker.image(baseImageName)
+                    image.pull()
+                } catch (e) {
+                    print e
+                    def baseImage  = docker.build(baseImageName, '--build-arg UID=`id -u` --build-arg GID=`id -g` .')
+                    baseImage.push dockerTag
+                }
+        }
+    }
+
+    def args = "-u 0:0 -v ${pwd}/artifacts:/artifacts:rw"
```

Comment:
I would definitely discourage using root user in docker container. In this way all artifact will have root rights and you wont be able to work with them outside of docker image without a sudo.